sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -413,6 +418,19 @@ static int sof_audio_probe(struct platform_device *pdev)
 
 	snd_soc_card_set_drvdata(&sof_audio_card_pcm512x, ctx);
 
+	/*
+	 * Enable GPIO4 for LED
+	 * we cannot use devm_gpiod_get since the device is NULL
+	 */
+	ctx->gpio_4 = gpiod_get(NULL, "PCM512x-GPIO4",
+				GPIOD_OUT_LOW);
+
+	/* don't stop on errors, just log messages */
+	if (IS_ERR_OR_NULL(ctx->gpio_4))
+		dev_err(&pdev->dev, "gpio4 not found\n");
```

Comment:
in fact I'm not sure this is consistent with the previous patch. The codec driver probing now would fail if it couldn't register the GPIO controller. So, if we cannot get any of the GPIOs it really means something bad is going on? In which case we can fail here too? Or we shouldn't fail probing in the codec driver if GPIO chip registration fails.

Reply:
In this case it's not fatal if the GPIO4 cannot be referenced, this is only for an LED. But you're right that moving forward the GPIOs will be used to check the clocks so that would indeed be an indicator that the configuration is broken. I'll stop the probe and follow your suggestion.

Reply:
And perhaps use *gpiod_get_optional() ?